### ISSUE TITLE
Elasticsearch sink does not recover from failure to create index template

### ISSUE THREAD

**Issue by nblumhardt** [serilog/serilog-sinks-elasticsearch#28]
Moved from: https://github.com/serilog/serilog/issues/453

> When a template is created (code), it can fail. If it fails, all successive indices created will not get the intended template settings.

**Comment by andreamaruccia** [comment]
Today had to set AutoRegisterTemplate = false because elsewise it would cause a total crash. Even in try-catch block it couldn't recover from it. Version is 3.0.134.

**Comment by mivano** [comment]
And what did you experience? What kind of crash?

**Comment by andreamaruccia** [comment]
Here is some context:

- implemented a wndows service with topshelf
- registered serilog with AutoRegisterTemplate = true
- the elastic search server is from elastic cloud (SaaS)

When I started the service (deployment with Octopus deploy) the service logged a log message before the logger creation but *nothing* after that. The process died without any successive log messages and Octoups reported simply: cannot start service.
I also checked eventlogs without luck. The only thing that worked is to remove AutoRegisterTemplate. No problems since then.
